src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs

```diff
@@ -146,7 +146,7 @@ public void Equals_Object_ReturnsExpected(GCHandle handle, object other, bool ex
             finally
             {
                 handle.Free();
-                if (other is GCHandle otherHandle)
+                if (other is GCHandle otherHandle && !expected)
```

Comment:
Oh bummer that's a good point. I actually just remembered I did this in the PR that introduced it and Stephen mentioned the same point about boxing, so I removed it without adding any other checks to prevent double frees. This is fine then :)